Update the DisplayName for Company Portal in Fleet UI

allenhouchins opened the pull request:
This pull request introduces a normalization utility to ensure consistent and user-friendly display of software names across the UI, specifically addressing problematic names like "Microsoft.CompanyPortal". The normalization logic is applied in multiple components and modals, improving clarity for end users. Additionally, icon mapping is updated to ensure the correct icon is shown for normalized names.

**Display Name Normalization:**

* Added a `normalizeSoftwareDisplayName` utility function in several components (e.g., `SoftwareNameCell`, `DeleteSoftwareModal`, `SoftwareDetailsSummary`) to map known problematic software names (such as "Microsoft.CompanyPortal") to more user-friendly versions ("Company Portal"). This normalization is now consistently applied wherever software names are displayed. [[1]](diffhunk://#diff-237686d473e6ad4a08d162138be8124bd53467b6964c70fd11c949aaafa77271R15-R26) [[2]](diffhunk://#diff-ed29a72570b7bc20e39604e1cf8d057e968cdffa25bd3f2ad736d42f382627d4R14-R26) [[3]](diffhunk://#diff-0ccef02e65bfd1b52964f0f4c729a1024c9832f6a6aba812bb770bf52b93bc37R86-R97)
* Updated all relevant UI components to use the normalized display name instead of the raw name, ensuring a consistent user experience in tables, modals, and summary cards. [[1]](diffhunk://#diff-237686d473e6ad4a08d162138be8124bd53467b6964c70fd11c949aaafa77271R190-R195) [[2]](diffhunk://#diff-237686d473e6ad4a08d162138be8124bd53467b6964c70fd11c949aaafa77271L191-R204) [[3]](diffhunk://#diff-237686d473e6ad4a08d162138be8124bd53467b6964c70fd11c949aaafa77271L211-R224) [[4]](diffhunk://#diff-ed29a72570b7bc20e39604e1cf8d057e968cdffa25bd3f2ad736d42f382627d4R54-R60) [[5]](diffhunk://#diff-ed29a72570b7bc20e39604e1cf8d057e968cdffa25bd3f2ad736d42f382627d4L77-R97) [[6]](diffhunk://#diff-ed29a72570b7bc20e39604e1cf8d057e968cdffa25bd3f2ad736d42f382627d4L90-R111) [[7]](diffhunk://#diff-0ccef02e65bfd1b52964f0f4c729a1024c9832f6a6aba812bb770bf52b93bc37R154-R156) [[8]](diffhunk://#diff-0ccef02e65bfd1b52964f0f4c729a1024c9832f6a6aba812bb770bf52b93bc37L212-R231)

**Icon Mapping Improvements:**

* Updated the `SOFTWARE_NAME_TO_ICON_MAP` and icon selection logic to recognize both "company portal" and "microsoft.companyportal" as valid keys for the Intune Company Portal icon, ensuring the correct icon is displayed regardless of the name format. [[1]](diffhunk://#diff-628095892e1d16090be1db6cc1a5c9cebc65248c32a8b1312385394818f2907bR273) [[2]](diffhunk://#diff-628095892e1d16090be1db6cc1a5c9cebc65248c32a8b1312385394818f2907bR519-R520)

allenhouchins committed:
Fix Company Portal display name

allenhouchins committed:
Update SoftwareNameCell.tsx

allenhouchins committed:
updates

allenhouchins committed:
Update DeleteSoftwareModal.tsx

jahzielv commented:
cc @marko-lisica for bringing this in

RachelElysia commented on frontend/components/TableContainer/DataTable/SoftwareNameCell/SoftwareNameCell.tsx:
```diff
@@ -12,6 +12,18 @@ import TooltipTruncatedTextCell from "../TooltipTruncatedTextCell";
 
 const baseClass = "software-name-cell";
 
+/**
```
Let's move this into `frontend/pages/SoftwarePage/helpers.tsx` and export it.

Then we can import it into each of the use cases. <3

RachelElysia commented on frontend/components/TableContainer/DataTable/SoftwareNameCell/SoftwareNameCell.tsx:
```diff
@@ -12,6 +12,18 @@ import TooltipTruncatedTextCell from "../TooltipTruncatedTextCell";
+  }
+  return name;
+};
+
```
One issue, I can see is if someone searches "Company Portal", it will come up empty because it uses the API which has it stored as "Microsoft.CompanyPortal".

Food for thought on next iteration to change this to a BE change for `display_name` and get the search by `display_name` running in the BE.

allenhouchins committed:
Move code to helpers file

allenhouchins committed:
Merge branch 'main' into allenhouchins-fix-company-portal-displayname

allenhouchins committed:
Revert "Move code to helpers file"

This reverts commit fb58650afaeeccf078d873e1a837743c8490d719.

allenhouchins committed:
Move normalizeSoftwareDisplayName to shared helpers

Consolidate normalizeSoftwareDisplayName function that was duplicated in three files (SoftwareDetailsSummary, DeleteSoftwareModal, and SoftwareNameCell) into a shared helper module.

allenhouchins commented on frontend/pages/SoftwarePage/helpers.tsx:
```diff
@@ -170,6 +170,8 @@ export const CUSTOM_TARGET_OPTIONS: IDropdownOption[] = [
   },
 ];
 
+export const SOFTWARE_NAME_CELL_BASE_CLASS = "software-name-cell";
```
@RachelElysia 🤦sigh... it's so obvious now. Changes have been implemented. Thanks for your patience with me!

lukeheath reviewed (changes requested):
Thanks for the contribution! 

This needs to go through product group drafting before it can be reviewed and merged. We welcome contributions from everyone, of course, but treat ICs from outside the product group as [community contributors](https://fleetdm.com/handbook/engineering#review-a-community-pull-request).
